moby/moby
# Enabling ILB/ELB on windows using per-node, per-network LB endpoint.

Note: This is 1 of 3 PRs (the other 2 are in libnetwork and swarmkit repos).

Signed-off-by: Pradip Dhara <pradipd@microsoft.com>

**- What I did**
Enabled ILB/ELB (routing mesh) functionality for windows.

**- How I did it**
This change enables ILB/ELB functionality on windows using a per-node, per-network LB endpoint.  The per-node, per-network LB endpoint was discussed with docker 
 as a solution to https://github.com/moby/moby/issues/30820.  Note that only windows is using the LB endpoint.  We have not changed the linux networking code to take advantage of the LB endpoint.


**- How to verify it**
make test on linux (note: I'm still seeing 2 tests occasionally fail.  But, I'm fairly certain they are unrelated to my change.  I'm sending this out now and will continue to debug).
We have added a bunch of powershell scripts to test out functionality. However, they are not public.  We will work with docker folks in figuring out how to make them public.  Ideally, we should get the docker_cli_swarm_tests running on windows, but, that is much bigger task and requires changes in windows.  

**- Description for the changelog**
Enabling ILB/ELB on windows using per-node, per-network LB endpoint.

**- A picture of a cute animal (not mandatory but encouraged)**
![tubby shaved](https://user-images.githubusercontent.com/14915188/29848109-93c9da6a-8cd3-11e7-84c2-7379cbd94fd3.jpg)
